base/stat.jl

```diff
@@ -163,3 +166,91 @@ function ismount(path...)
     (s1.inode == s2.inode) && return true
     false
 end
+
+# Cross-platform case-sensitive path canonicalization.
+# _ispath_casesensitive(path) assumes that ispath(path) is true,
+# and returns whether path matches the case stored in the filesystem.
+if is_unix() && !is_apple()
+    # assume case-sensitive filesystems, don't have to do anything
+    _ispath_casesensitive(path) = true
+elseif is_windows()
+    # GetLongPathName Win32 function returns the case-preserved filename on NTFS.
+    _ispath_casesensitive(path) = Filesystem.longpath(path) == path
```

Comment:
Okay, I just checked, and `longpath(path)` on Windows seems to leave trailing slashes as-is, so we are fine here.